src/file_arc.c

```diff
@@ -229,154 +231,202 @@ clist = data->cores;
-			data->pbc[2] = 1.0;
-			data->pbc[3] = 0.5*G_PI;
-			data->pbc[4] = 0.5*G_PI;
-			data->periodic = 2;
-		}
-    }else{/*less than 6 tokens -> PCB=2D */
-		data->pbc[5] = D2R*(data->pbc[2]);
-		data->pbc[2] = 1.0;
-		data->pbc[3] = 0.5*G_PI;
-		data->pbc[4] = 0.5*G_PI;
-    }
+    if((data->pbc[5])>=0)
+      { /* 6 tokens -> PCB=ON with or without implicit 2D */
+	data->pbc[3] *= D2R;
+	data->pbc[4] *= D2R;
+	data->pbc[5] *= D2R;
+	if((data->pbc[2])==0)
+          { /* implicit 2D */
+	  data->pbc[2] = 1.0;
+	  data->pbc[3] = 0.5*G_PI;
+	  data->pbc[4] = 0.5*G_PI;
+	  data->periodic = 2;
+	  }
+        }
+      else
+        { /* less than 6 tokens -> PCB=2D */
+	data->pbc[5] = D2R*(data->pbc[2]);
+	data->pbc[2] = 1.0;
+	data->pbc[3] = 0.5*G_PI;
+	data->pbc[4] = 0.5*G_PI;
+        }
+      matrix_lattice_init(data); /* initialize matrix */
 #if DEBUG_ARC
-fprintf(stdout,"#DBG: a=%lf b=%lf c=%lf alpha=%lf beta=%lf gamma=%lf\n",data->pbc[0],data->pbc[1],data->pbc[2],data->pbc[3],data->pbc[4],data->pbc[5]);
+fprintf(stdout, "#DBG: a=%lf b=%lf c=%lf alpha=%lf beta=%lf gamma=%lf\n", 
+	data->pbc[0], data->pbc[1], data->pbc[2], data->pbc[3], data->pbc[4], data->pbc[5]);
+      P3MAT("lattice matrix", model->latmat);
 #endif
-    matrix_lattice_init(data);/*initialize matrix now*/
-    data->fractional = TRUE;/*set fractional now*/
-    g_free(line);
-    continue;
-  }
+      data->fractional = TRUE; /* set fractional */
+      g_free(line);
+      continue;
+    }
 
-/* each coord line is defined as:
+/* each coord line consists of
  * [name], x, y, z, type, [seq. nb], pot. type, elem symbol, charge [, atom nb] */
-  c_symb[0]='-';c_symb[1]='\0';
-  sscanf(line,"%*s %lf %lf %lf %s %*s %s %s %lf %*s",&x,&y,&z,&(c_type[0]),&(c_ptype[0]),&(c_symb[0]),&charge);
+    c_symb[0]='-';c_symb[1]='\0';
+    sscanf(line,"%*s %lf %lf %lf %s %*s %s %s %lf %*s", 
+		  &x, &y, &z, &(c_type[0]), &(c_ptype[0]), &(c_symb[0]), &charge);
 #if DEBUG_ARC
-fprintf(stdout,"#DBG: LINE: %lf %lf %lf %s %s %s %f\n",x,y,z,c_type,c_ptype,c_symb,charge);
+fprintf(stdout,"#DBG: LINE: %lf %lf %lf %s %s %s %f\n", x, y, z, c_type, c_ptype, c_symb, charge);
 #endif
-  if(c_symb[0]=='-'){
-	gui_text_show(ERROR, "unexpected end of file reading arc file\n");
-	g_free(line);
-	return;
-  }
+    if(c_symb[0]=='-')
+      {
+      gui_text_show(ERROR, "Unexpected end of file reading arc file\n");
+      g_free(line);
+      return;
+      }
 /* process core/shell candidates */
-  if (elem_symbol_test(c_symb)){
-	core_flag=TRUE;
-	region = 0;
+    if (elem_symbol_test(c_symb))
+      {
+      core_flag=TRUE;
+      region = 0;
 /* MARVIN core/shell/region parse */
-	if (is_marvin_label(c_type)){
-		if (!marvin_core(c_type)) core_flag = FALSE;
-		region = marvin_region(c_type);
-		data->region_empty[region] = FALSE;
-	}
+      if (is_marvin_label(c_type))
+        {
+        if (!marvin_core(c_type)) core_flag = FALSE;
+          region = marvin_region(c_type);
```

Comment:
Indentation problem.
Also, we don't have a C styling policy (that I know of).
So the many changes that only add a space or tab here and there are just making it harder to find what is really changed ;)